In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
#!pip install pyarrow

In [3]:
# for year in range(3, 24):
#     # Formating the year to have two digits
#     year_str = f"{year:02d}"
    
#     # Parquet path
#     parquet_file = f'malaria_notification_data/NOTIPO{year_str}.parquet'
    
#     # CSV path
#     csv_file = f'malaria_notification_data/NOTIPO{year_str}.csv'
    
#     parquet_df = pd.read_parquet(parquet_file)
#     parquet_df.to_csv(csv_file, index=False)

In [2]:
malaria_2003_df = pd.read_csv('malaria_notification_data_csv/NOTIPO03.csv')
malaria_2004_df = pd.read_csv('malaria_notification_data_csv/NOTIPO04.csv')
malaria_2005_df = pd.read_csv('malaria_notification_data_csv/NOTIPO05.csv')
malaria_2006_df = pd.read_csv('malaria_notification_data_csv/NOTIPO06.csv')
malaria_2007_df = pd.read_csv('malaria_notification_data_csv/NOTIPO07.csv')
malaria_2008_df = pd.read_csv('malaria_notification_data_csv/NOTIPO08.csv')
malaria_2009_df = pd.read_csv('malaria_notification_data_csv/NOTIPO09.csv')
malaria_2010_df = pd.read_csv('malaria_notification_data_csv/NOTIPO10.csv')
malaria_2011_df = pd.read_csv('malaria_notification_data_csv/NOTIPO11.csv')
malaria_2012_df = pd.read_csv('malaria_notification_data_csv/NOTIPO12.csv')
malaria_2013_df = pd.read_csv('malaria_notification_data_csv/NOTIPO13.csv')
malaria_2014_df = pd.read_csv('malaria_notification_data_csv/NOTIPO14.csv')
malaria_2015_df = pd.read_csv('malaria_notification_data_csv/NOTIPO15.csv')
malaria_2016_df = pd.read_csv('malaria_notification_data_csv/NOTIPO16.csv')
malaria_2017_df = pd.read_csv('malaria_notification_data_csv/NOTIPO17.csv')
malaria_2018_df = pd.read_csv('malaria_notification_data_csv/NOTIPO18.csv')
malaria_2019_df = pd.read_csv('malaria_notification_data_csv/NOTIPO19.csv')
malaria_2020_df = pd.read_csv('malaria_notification_data_csv/NOTIPO20.csv')
malaria_2021_df = pd.read_csv('malaria_notification_data_csv/NOTIPO21.csv')
malaria_2022_df = pd.read_csv('malaria_notification_data_csv/NOTIPO22.csv')
malaria_2023_df = pd.read_csv('malaria_notification_data_csv/NOTIPO23.csv')

C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (30) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (30,41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (37,38,41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,

In [3]:
full_malaria_df = pd.concat([
    malaria_2003_df, malaria_2004_df, malaria_2005_df, malaria_2006_df,
    malaria_2007_df, malaria_2008_df, malaria_2009_df, malaria_2010_df,
    malaria_2011_df, malaria_2012_df, malaria_2013_df, malaria_2014_df,
    malaria_2015_df, malaria_2016_df, malaria_2017_df, malaria_2018_df,
    malaria_2019_df, malaria_2020_df, malaria_2021_df, malaria_2022_df,
    malaria_2023_df])

full_malaria_df = full_malaria_df.reset_index().drop(columns=['index', 'Unnamed: 0'])
full_malaria_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM  \
0              56  27/01/03  23/03/2011       NaN  05/2003         3   
1             470  20/03/03  23/03/2011       NaN  12/2003         2   
2            1309  10/07/03  23/03/2011       NaN  28/2003         1   
3              19  27/01/03  23/03/2011       NaN  05/2003         3   
4             657  04/04/03  23/03/2011       NaN  14/2003         1   
...           ...       ...         ...       ...      ...       ...   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1   
6816640        14  26/09/23  20/10/2023  20/10/23  39/2023         1   
6816641         1  28/08/23  11/03/2024  11/03/24  35/2023         2   

         UF_NOTIF  MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  \
0              21    210290      9505  498068.0  ...                  NaN   
1              21    210232     10033  232514.0  ...                  NaN   
2              21    210232     10032       NaN  ...                  NaN   
3              21    210232     10032  232516.0  ...                  NaN   
4              21    210232    100321       NaN  ...                  NaN   
...           ...       ...       ...       ...  ...                  ...   
6816637        13    130020        49     618.0  ...                  NaN   
6816638        13    130406        46      64.0  ...                  NaN   
6816639        13    130406        24     339.0  ...                  NaN   
6816640        16    160025         1  352826.0  ...                  NaN   
6816641        17    172120   1721201    1789.0  ...                  NaN   

         EXAMINADOR ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  \
0               NaN       1         5          NaN           NaN   
1               NaN       2         6          NaN           NaN   
2               NaN       2       NaN          NaN           NaN   
3               NaN       1         6          NaN           NaN   
4               NaN       2         4          NaN           NaN   
...             ...     ...       ...          ...           ...   
6816637       618.0       2         6          NaN           NaN   
6816638        67.0       2         6          NaN           NaN   
6816639       449.0       2         6          NaN           NaN   
6816640    352826.0       2         6          NaN           NaN   
6816641      1789.0       2         6          NaN           NaN   

         DIMEA_AMAMENT  VIVAX  FALCIPARUM  NIV_ESCO  
0                  NaN    NaN         NaN       0.0  
1                  NaN    NaN         NaN       1.0  
2                  NaN    NaN         NaN       0.0  
3                  NaN    NaN         NaN       NaN  
4                  NaN    NaN         NaN       0.0  
...                ...    ...         ...       ...  
6816637            NaN    2.0         2.0       1.0  
6816638            NaN    2.0         2.0       0.0  
6816639            NaN    2.0         2.0       2.0  
6816640            NaN    2.0         2.0       5.0  
6816641            NaN    2.0         2.0       6.0  

[6816642 rows x 48 columns]

#### Gets vivax cases (RES_EXAM = 4)

In [4]:
vivax_cases_df = full_malaria_df[full_malaria_df.get('RES_EXAM')==4]
vivax_cases_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM  \
0              56  27/01/03  23/03/2011       NaN  05/2003         3   
1             470  20/03/03  23/03/2011       NaN  12/2003         2   
2            1309  10/07/03  23/03/2011       NaN  28/2003         1   
3              19  27/01/03  23/03/2011       NaN  05/2003         3   
4             657  04/04/03  23/03/2011       NaN  14/2003         1   
...           ...       ...         ...       ...      ...       ...   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1   
6816640        14  26/09/23  20/10/2023  20/10/23  39/2023         1   
6816641         1  28/08/23  11/03/2024  11/03/24  35/2023         2   

         UF_NOTIF  MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  \
0              21    210290      9505  498068.0  ...                  NaN   
1              21    210232     10033  232514.0  ...                  NaN   
2              21    210232     10032       NaN  ...                  NaN   
3              21    210232     10032  232516.0  ...                  NaN   
4              21    210232    100321       NaN  ...                  NaN   
...           ...       ...       ...       ...  ...                  ...   
6816637        13    130020        49     618.0  ...                  NaN   
6816638        13    130406        46      64.0  ...                  NaN   
6816639        13    130406        24     339.0  ...                  NaN   
6816640        16    160025         1  352826.0  ...                  NaN   
6816641        17    172120   1721201    1789.0  ...                  NaN   

         EXAMINADOR ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  \
0               NaN       1         5          NaN           NaN   
1               NaN       2         6          NaN           NaN   
2               NaN       2       NaN          NaN           NaN   
3               NaN       1         6          NaN           NaN   
4               NaN       2         4          NaN           NaN   
...             ...     ...       ...          ...           ...   
6816637       618.0       2         6          NaN           NaN   
6816638        67.0       2         6          NaN           NaN   
6816639       449.0       2         6          NaN           NaN   
6816640    352826.0       2         6          NaN           NaN   
6816641      1789.0       2         6          NaN           NaN   

         DIMEA_AMAMENT  VIVAX  FALCIPARUM  NIV_ESCO  
0                  NaN    NaN         NaN       0.0  
1                  NaN    NaN         NaN       1.0  
2                  NaN    NaN         NaN       0.0  
3                  NaN    NaN         NaN       NaN  
4                  NaN    NaN         NaN       0.0  
...                ...    ...         ...       ...  
6816637            NaN    2.0         2.0       1.0  
6816638            NaN    2.0         2.0       0.0  
6816639            NaN    2.0         2.0       2.0  
6816640            NaN    2.0         2.0       5.0  
6816641            NaN    2.0         2.0       6.0  

[5508409 rows x 48 columns]

In [5]:
len(vivax_cases_df)/len(full_malaria_df)*100

80.80824840148566

About 81% of all malaria cases between 2023 and 2023 were caused by P. Vivax

#### Gets name of UFs

In [6]:
uf_dict = {11 : 'RO',
           12 : 'AC',
           13 : 'AM',
           14 : 'RR',
           15 : 'PA',
           16 : 'AP',
           17 : 'TO',
           21 : 'MA',
           22 : 'PI',
           23 : 'CE',
           24 : 'RN',
           25 : 'PB',
           26 : 'PE',
           27 : 'AL',
           28 : 'SE',
           29 : 'BA',
           31 : 'MG', 
           32 : 'ES',
           33 : 'RJ',
           35 : 'SP',
           41 : 'PR',
           42 : 'SC',
           43 : 'RS',
           50 : 'MS',
           51 : 'MT',
           52 : 'GO',
           53 : 'DF'
          }
corrected_ufs_df = vivax_cases_df.replace({"UF_NOTIF": uf_dict, "UF_RESID": uf_dict})
corrected_ufs_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0              56  27/01/03  23/03/2011       NaN  05/2003         3       MA   
1             470  20/03/03  23/03/2011       NaN  12/2003         2       MA   
2            1309  10/07/03  23/03/2011       NaN  28/2003         1       MA   
3              19  27/01/03  23/03/2011       NaN  05/2003         3       MA   
4             657  04/04/03  23/03/2011       NaN  14/2003         1       MA   
...           ...       ...         ...       ...      ...       ...      ...   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1       AM   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1       AM   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1       AM   
6816640        14  26/09/23  20/10/2023  20/10/23  39/2023         1       AP   
6816641         1  28/08/23  11/03/2024  11/03/24  35/2023         2       TO   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          210290      9505  498068.0  ...                  NaN         NaN   
1          210232     10033  232514.0  ...                  NaN         NaN   
2          210232     10032       NaN  ...                  NaN         NaN   
3          210232     10032  232516.0  ...                  NaN         NaN   
4          210232    100321       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816637    130020        49     618.0  ...                  NaN       618.0   
6816638    130406        46      64.0  ...                  NaN        67.0   
6816639    130406        24     339.0  ...                  NaN       449.0   
6816640    160025         1  352826.0  ...                  NaN    352826.0   
6816641    172120   1721201    1789.0  ...                  NaN      1789.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             1         5          NaN           NaN            NaN   NaN   
1             2         6          NaN           NaN            NaN   NaN   
2             2       NaN          NaN           NaN            NaN   NaN   
3             1         6          NaN           NaN            NaN   NaN   
4             2         4          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816637       2         6          NaN           NaN            NaN   2.0   
6816638       2         6          NaN           NaN            NaN   2.0   
6816639       2         6          NaN           NaN            NaN   2.0   
6816640       2         6          NaN           NaN            NaN   2.0   
6816641       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
0               NaN       0.0  
1               NaN       1.0  
2               NaN       0.0  
3               NaN       NaN  
4               NaN       0.0  
...             ...       ...  
6816637         2.0       1.0  
6816638         2.0       0.0  
6816639         2.0       2.0  
6816640         2.0       5.0  
6816641         2.0       6.0  

[5508409 rows x 48 columns]

#### Gets data for the Amazon region

In [7]:
amazon_df = corrected_ufs_df[corrected_ufs_df.get('UF_NOTIF').isin(['RO', 'AC', 'AM', 'RR', 'PA', 'AP', 'TO'])]
amazon_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1       AM   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1       AM   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1       AM   
6816640        14  26/09/23  20/10/2023  20/10/23  39/2023         1       AP   
6816641         1  28/08/23  11/03/2024  11/03/24  35/2023         2       TO   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816637    130020        49     618.0  ...                  NaN       618.0   
6816638    130406        46      64.0  ...                  NaN        67.0   
6816639    130406        24     339.0  ...                  NaN       449.0   
6816640    160025         1  352826.0  ...                  NaN    352826.0   
6816641    172120   1721201    1789.0  ...                  NaN      1789.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816637       2         6          NaN           NaN            NaN   2.0   
6816638       2         6          NaN           NaN            NaN   2.0   
6816639       2         6          NaN           NaN            NaN   2.0   
6816640       2         6          NaN           NaN            NaN   2.0   
6816641       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816637         2.0       1.0  
6816638         2.0       0.0  
6816639         2.0       2.0  
6816640         2.0       5.0  
6816641         2.0       6.0  

[5357677 rows x 48 columns]

In [8]:
amazonas_df = amazon_df[amazon_df.get('UF_NOTIF')=='AM']
amazonas_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   
6816624      2261  07/08/23  10/08/2023  10/08/23  32/2023         1       AM   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1       AM   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1       AM   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816622    130260       414     244.0  ...                 15,4      3160.0   
6816624    130020        20      29.0  ...                  NaN       460.0   
6816637    130020        49     618.0  ...                  NaN       618.0   
6816638    130406        46      64.0  ...                  NaN        67.0   
6816639    130406        24     339.0  ...                  NaN       449.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816622       2         6          NaN           NaN            NaN   2.0   
6816624       2         5          2.0           NaN            NaN   2.0   
6816637       2         6          NaN           NaN            NaN   2.0   
6816638       2         6          NaN           NaN            NaN   2.0   
6816639       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816622         2.0       6.0  
6816624         2.0       6.0  
6816637         2.0       1.0  
6816638         2.0       0.0  
6816639         2.0       2.0  

[2013685 rows x 48 columns]

In [9]:
len(amazonas_df)/len(vivax_cases_df)*100

36.55656288412861

About 36.5% of all malaria cases caused by P. Vivax between 2023 and 2023 were notified in the state of Amazonas

In [10]:
amazonas_df['UF_NOTIF'].unique()

array(['AM'], dtype=object)

In [11]:
amazonas_df['UF_RESID'].unique()

array(['AM', 'RO', 'MT', 'AC', 'PA', nan, 'RR', 'MA', 'RS', 'AP', 'ES',
       'DF', 'PI', 'MG', 'BA', 'TO', 'GO', 'PE', 'AL', 'PR', 'MS', 'SC',
       'SP', 'RJ', 'RN'], dtype=object)

In [12]:
amazonas_notif_and_resid = amazon_df[(amazon_df.get('UF_NOTIF')=='AM') & (amazon_df.get('UF_RESID')=='AM')]
amazonas_notif_and_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816621      1019  30/10/23  01/11/2023  01/11/23  44/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   
6816624      2261  07/08/23  10/08/2023  10/08/23  32/2023         1       AM   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1       AM   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816621    130060         4      57.0  ...                  NaN        11.0   
6816622    130260       414     244.0  ...                 15,4      3160.0   
6816624    130020        20      29.0  ...                  NaN       460.0   
6816638    130406        46      64.0  ...                  NaN        67.0   
6816639    130406        24     339.0  ...                  NaN       449.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816621       1         6          NaN           NaN            NaN   1.0   
6816622       2         6          NaN           NaN            NaN   2.0   
6816624       2         5          2.0           NaN            NaN   2.0   
6816638       2         6          NaN           NaN            NaN   2.0   
6816639       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816621         2.0       5.0  
6816622         2.0       6.0  
6816624         2.0       6.0  
6816638         2.0       0.0  
6816639         2.0       2.0  

[2004500 rows x 48 columns]

In [13]:
len(amazonas_notif_and_resid)/len(amazonas_df)*100

99.5438710622565

From all cases notified in the state of Amazonas, about 99.5% of the infected were residents of the state.

In [14]:
cases_by_AM_municipality = amazonas_notif_and_resid[['MUN_RESI']].groupby(['MUN_RESI']).size().reset_index(name='TOTAL_CASES').sort_values(by='TOTAL_CASES', ascending=False)
cases_by_AM_municipality['MUN_RESI'] = cases_by_AM_municipality['MUN_RESI'].astype(int)
cases_by_AM_municipality

MUN_RESI  TOTAL_CASES
37    130260       521362
51    130380       134459
7     130040        92387
19    130120        77098
57    130420        72717
..       ...          ...
58    130423          896
3     130010          505
8     130050          182
11    130068          177
61    130440          155

[62 rows x 2 columns]

It is possible to see that Manaus is the city of residence of most notified cases in the state of Amazonas

In [15]:
manaus_notif_and_resid = amazonas_notif_and_resid[(amazonas_notif_and_resid.get('MUN_NOTI')==130260) & (amazonas_notif_and_resid.get('MUN_RESI')==130260)]
manaus_notif_and_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816487      8400  20/12/23  21/12/2023  21/12/23  51/2023         2       AM   
6816566        95  26/09/23  09/10/2023  09/10/23  39/2023         1       AM   
6816585       130  24/12/23  28/12/2023  28/12/23  52/2023         1       AM   
6816608       125  17/12/23  21/12/2023  21/12/23  51/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816487    130260        19    2745.0  ...                 17,5        50.0   
6816566    130260       103    4317.0  ...                  NaN      4317.0   
6816585    130260       103    4468.0  ...                  NaN      4468.0   
6816608    130260       103    4913.0  ...                  NaN      4913.0   
6816622    130260       414     244.0  ...                 15,4      3160.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816487       2         6          NaN           NaN            NaN   2.0   
6816566       2         6          NaN           NaN            NaN   2.0   
6816585       2         6          NaN           NaN            NaN   2.0   
6816608       2         6          NaN           NaN            NaN   2.0   
6816622       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816487         2.0       5.0  
6816566         2.0      10.0  
6816585         2.0       1.0  
6816608         2.0       1.0  
6816622         2.0       6.0  

[516683 rows x 48 columns]

In [16]:
manaus_notif = amazonas_notif_and_resid[amazonas_notif_and_resid.get('MUN_NOTI')==130260]
manaus_notif

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816487      8400  20/12/23  21/12/2023  21/12/23  51/2023         2       AM   
6816566        95  26/09/23  09/10/2023  09/10/23  39/2023         1       AM   
6816585       130  24/12/23  28/12/2023  28/12/23  52/2023         1       AM   
6816608       125  17/12/23  21/12/2023  21/12/23  51/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816487    130260        19    2745.0  ...                 17,5        50.0   
6816566    130260       103    4317.0  ...                  NaN      4317.0   
6816585    130260       103    4468.0  ...                  NaN      4468.0   
6816608    130260       103    4913.0  ...                  NaN      4913.0   
6816622    130260       414     244.0  ...                 15,4      3160.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816487       2         6          NaN           NaN            NaN   2.0   
6816566       2         6          NaN           NaN            NaN   2.0   
6816585       2         6          NaN           NaN            NaN   2.0   
6816608       2         6          NaN           NaN            NaN   2.0   
6816622       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816487         2.0       5.0  
6816566         2.0      10.0  
6816585         2.0       1.0  
6816608         2.0       1.0  
6816622         2.0       6.0  

[522947 rows x 48 columns]

In [17]:
manaus_resid = amazonas_notif_and_resid[amazonas_notif_and_resid.get('MUN_RESI')==130260]
manaus_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816487      8400  20/12/23  21/12/2023  21/12/23  51/2023         2       AM   
6816566        95  26/09/23  09/10/2023  09/10/23  39/2023         1       AM   
6816585       130  24/12/23  28/12/2023  28/12/23  52/2023         1       AM   
6816608       125  17/12/23  21/12/2023  21/12/23  51/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816487    130260        19    2745.0  ...                 17,5        50.0   
6816566    130260       103    4317.0  ...                  NaN      4317.0   
6816585    130260       103    4468.0  ...                  NaN      4468.0   
6816608    130260       103    4913.0  ...                  NaN      4913.0   
6816622    130260       414     244.0  ...                 15,4      3160.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816487       2         6          NaN           NaN            NaN   2.0   
6816566       2         6          NaN           NaN            NaN   2.0   
6816585       2         6          NaN           NaN            NaN   2.0   
6816608       2         6          NaN           NaN            NaN   2.0   
6816622       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816487         2.0       5.0  
6816566         2.0      10.0  
6816585         2.0       1.0  
6816608         2.0       1.0  
6816622         2.0       6.0  

[521362 rows x 48 columns]

In [18]:
manaus_resid['YEAR_NOTIF'] = manaus_resid['DT_NOTIF'].str[-2:].astype(int)
manaus_resid['YEAR_NOTIF'] = manaus_resid['YEAR_NOTIF'].apply(lambda x: 2000 + x if x <= 23 else 1900 + x)
manaus_resid

<ipython-input-18-bb7fdcbcbd17>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  manaus_resid['YEAR_NOTIF'] = manaus_resid['DT_NOTIF'].str[-2:].astype(int)
<ipython-input-18-bb7fdcbcbd17>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  manaus_resid['YEAR_NOTIF'] = manaus_resid['YEAR_NOTIF'].apply(lambda x: 2000 + x if x <= 23 else 1900 + x)


COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816487      8400  20/12/23  21/12/2023  21/12/23  51/2023         2       AM   
6816566        95  26/09/23  09/10/2023  09/10/23  39/2023         1       AM   
6816585       130  24/12/23  28/12/2023  28/12/23  52/2023         1       AM   
6816608       125  17/12/23  21/12/2023  21/12/23  51/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... EXAMINADOR  ID_LVC  GESTANTE   \
13         130260        46     653.0  ...        NaN        2         6   
16         130260        46     647.0  ...        NaN        2         6   
17         130260        46     647.0  ...        NaN        2         6   
18         130260        44    1281.0  ...        NaN        2         5   
19         130260        44       NaN  ...        NaN        2       NaN   
...           ...       ...       ...  ...        ...      ...       ...   
6816487    130260        19    2745.0  ...       50.0        2         6   
6816566    130260       103    4317.0  ...     4317.0        2         6   
6816585    130260       103    4468.0  ...     4468.0        2         6   
6816608    130260       103    4913.0  ...     4913.0        2         6   
6816622    130260       414     244.0  ...     3160.0        2         6   

        AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT  VIVAX FALCIPARUM  NIV_ESCO  \
13              NaN           NaN            NaN    NaN        NaN       1.0   
16              NaN           NaN            NaN    NaN        NaN       1.0   
17              NaN           NaN            NaN    NaN        NaN       1.0   
18              NaN           NaN            NaN    NaN        NaN       3.0   
19              NaN           NaN            NaN    NaN        NaN      10.0   
...             ...           ...            ...    ...        ...       ...   
6816487         NaN           NaN            NaN    2.0        2.0       5.0   
6816566         NaN           NaN            NaN    2.0        2.0      10.0   
6816585         NaN           NaN            NaN    2.0        2.0       1.0   
6816608         NaN           NaN            NaN    2.0        2.0       1.0   
6816622         NaN           NaN            NaN    2.0        2.0       6.0   

         YEAR_NOTIF  
13             2003  
16             2003  
17             2003  
18             2003  
19             2003  
...             ...  
6816487        2023  
6816566        2023  
6816585        2023  
6816608        2023  
6816622        2023  

[521362 rows x 49 columns]

In [19]:
cases_Manaus_by_year = manaus_resid[['YEAR_NOTIF']].groupby(['YEAR_NOTIF']).size().reset_index(name='TOTAL_CASES').sort_values(by='TOTAL_CASES', ascending=False)
cases_Manaus_by_year['YEAR_NOTIF'] = cases_Manaus_by_year['YEAR_NOTIF'].astype(int)
cases_Manaus_by_year

YEAR_NOTIF  TOTAL_CASES
0         2003        83665
2         2005        72327
1         2004        64223
4         2007        52405
3         2006        48769
5         2008        25227
6         2009        21081
7         2010        19892
8         2011        18593
14        2017        16028
9         2012        13138
15        2018        12593
13        2016        12120
12        2015        11820
16        2019         9699
11        2014         8579
10        2013         8249
17        2020         7490
18        2021         6155
20        2023         5226
19        2022         4083

In [20]:
cases_Manaus_by_year.sort_values(by='YEAR_NOTIF', ascending=True)

YEAR_NOTIF  TOTAL_CASES
0         2003        83665
1         2004        64223
2         2005        72327
3         2006        48769
4         2007        52405
5         2008        25227
6         2009        21081
7         2010        19892
8         2011        18593
9         2012        13138
10        2013         8249
11        2014         8579
12        2015        11820
13        2016        12120
14        2017        16028
15        2018        12593
16        2019         9699
17        2020         7490
18        2021         6155
19        2022         4083
20        2023         5226

In [21]:
vivax_cases_to_csv = vivax_cases_df

In [22]:
vivax_cases_to_csv['PAIS_INF'].unique()

array([  1., 216.,  84., 198.,  85.,  nan,  25.,   3., 165., 126.,  72.,
        48.,   2., 183., 999., 163., 160.,  27., 188.,  86., 178., 182.,
       193., 175., 195., 169., 199., 128.,  73.,  39., 154.,  36.,  82.,
       179.,  91.,  76.,  35., 140., 194.,  63., 170.,   7., 113., 172.,
        64., 110., 112.,  50.,  55.,  74., 150., 162.,   9.,  13., 224.,
         8.,  56., 173.,  66.,  17.,  28.,  18.,  30.,  32.,  97.,  88.,
       213.,  31.,  19., 217.,  42.,  90.,  71.,   5., 200.,  75.,  98.,
       104., 171.,  34.])

Dropping cases where the infection occured outside of Brazil

In [23]:
vivax_cases_to_csv = vivax_cases_to_csv[vivax_cases_to_csv.get('PAIS_INF')==1]

Removes unused columns

In [24]:
vivax_cases_to_csv.columns

Index(['COD_NOTI', 'DT_NOTIF', 'DT_ENVLO', 'DT_DIGIT', 'SEM_NOTI', 'TIPO_LAM',
       'UF_NOTIF', 'MUN_NOTI', 'COD_UNIN', 'COD_AGEN', 'DT_NASCI', 'ID_PACIE',
       'ID_DIMEA', 'SEXO', 'RACA', 'PESO', 'PAIS_RES', 'UF_RESID', 'MUN_RESI',
       'LOC_RESI', 'SINTOMAS', 'DT_SINTO', 'COD_OCUP', 'UF_INFEC', 'PAIS_INF',
       'MUN_INFE', 'LOC_INFE', 'DT_EXAME', 'RES_EXAM', 'QTD_PARA', 'QTD_CRUZ',
       'DT_TRATA', 'ESQUEMA', 'CO_FORMA_SANGUINEA',
       'CO_FORMA_TECIDUAL_GAMETOCITO', 'HEMOPARASI', 'EXAME ',
       'NU_ATIVIDADE_G6PD', 'NU_HEMOGLOBINA_TOTAL', 'EXAMINADOR', 'ID_LVC ',
       'GESTANTE ', 'AMAMENTANDO', 'TEMP_AMAMENT', 'DIMEA_AMAMENT', 'VIVAX',
       'FALCIPARUM', 'NIV_ESCO'],
      dtype='object')

In [28]:
vivax_dropped_columns = vivax_cases_to_csv.drop(columns=['DT_TRATA', 'ESQUEMA', 'CO_FORMA_SANGUINEA', 
                                'CO_FORMA_TECIDUAL_GAMETOCITO', 'EXAME ', 'NU_ATIVIDADE_G6PD', 
                                'NU_HEMOGLOBINA_TOTAL', 'EXAMINADOR', 'ID_LVC ', 'GESTANTE ', 
                                'AMAMENTANDO', 'TEMP_AMAMENT', 'DIMEA_AMAMENT', 'NIV_ESCO', 'FALCIPARUM',
                                'QTD_PARA', 'QTD_CRUZ', 'SEXO', 'RACA', 'PESO', 'COD_AGEN', 'DT_NASCI', 'ID_PACIE',
                                'ID_DIMEA', 'COD_UNIN', 'COD_NOTI', 'TIPO_LAM', 'DT_ENVLO', 'HEMOPARASI', 
                                'PAIS_INF', 'DT_DIGIT'])
vivax_dropped_columns

DT_NOTIF SEM_NOTI  UF_NOTIF  MUN_NOTI  PAIS_RES  UF_RESID  MUN_RESI  \
0        27/01/03  05/2003        21    210290       1.0      21.0  210055.0   
1        20/03/03  12/2003        21    210232       1.0      21.0  210232.0   
2        10/07/03  28/2003        21    210232       1.0      21.0  210203.0   
3        27/01/03  05/2003        21    210232       1.0      21.0  210203.0   
4        04/04/03  14/2003        21    210232       1.0      21.0  210203.0   
...           ...      ...       ...       ...       ...       ...       ...   
6816627  21/09/23  38/2023        21    210177       1.0      21.0  210177.0   
6816628  13/10/23  41/2023        12    120045       1.0      13.0  130070.0   
6816632  25/10/23  43/2023        14    140015       1.0      14.0  140010.0   
6816633  29/09/23  39/2023        16    160060       1.0      16.0  160060.0   
6816640  26/09/23  39/2023        16    160025       1.0      16.0  160025.0   

         LOC_RESI  SINTOMAS  DT_SINTO  COD_OCUP  UF_INFEC  MUN_INFE  LOC_INFE  \
0            29.0         1  25/01/03      99.0      21.0  210055.0      29.0   
1             NaN         1  18/03/03      11.0      21.0  210203.0      41.0   
2            55.0         1  06/07/03       1.0      21.0  210203.0      55.0   
3            55.0         1  26/01/03       1.0      21.0  210203.0      55.0   
4             NaN         1  01/04/03       3.0      21.0  210203.0      88.0   
...           ...       ...       ...       ...       ...       ...       ...   
6816627       5.0         1  20/09/23       5.0      15.0  150360.0       NaN   
6816628       NaN         1  11/01/23       7.0      13.0  130070.0       NaN   
6816632    1148.0         1  23/10/23       2.0      14.0  140005.0       NaN   
6816633      32.0         1  26/09/23      11.0      16.0  160027.0       NaN   
6816640      30.0         1  23/09/23      11.0      16.0  160053.0       NaN   

         DT_EXAME  RES_EXAM  VIVAX  
0        27/01/03         4    NaN  
1        20/03/03         4    NaN  
2        10/07/03         4    NaN  
3        27/01/03         4    NaN  
4        04/04/03         4    NaN  
...           ...       ...    ...  
6816627  21/09/23         4    1.0  
6816628  13/10/23         4    1.0  
6816632  25/10/23         4    1.0  
6816633  29/09/23         4    2.0  
6816640  26/09/23         4    2.0  

[5401634 rows x 17 columns]

In [26]:
#vivax_cases_to_csv.to_csv('malaria_notification_data_csv/Vivax_cases.csv', index=False)

In [29]:
vivax_dropped_columns.columns

Index(['DT_NOTIF', 'SEM_NOTI', 'UF_NOTIF', 'MUN_NOTI', 'PAIS_RES', 'UF_RESID',
       'MUN_RESI', 'LOC_RESI', 'SINTOMAS', 'DT_SINTO', 'COD_OCUP', 'UF_INFEC',
       'MUN_INFE', 'LOC_INFE', 'DT_EXAME', 'RES_EXAM', 'VIVAX'],
      dtype='object')

Dropping rows with NaN values

In [34]:
vivax_dropped_columns_dropna = vivax_dropped_columns.dropna()
vivax_dropped_columns_dropna

DT_NOTIF SEM_NOTI  UF_NOTIF  MUN_NOTI  PAIS_RES  UF_RESID  MUN_RESI  \
325201   23/06/03  26/2003        11    110013       1.0      11.0  110013.0   
1793840  01/02/06  05/2006        11    110020       1.0      11.0  110020.0   
2373846  25/05/06  21/2006        11    110160       1.0      11.0  110160.0   
2488087  24/12/07  52/2007        11    110020       1.0      11.0  110020.0   
2530206  02/01/07  01/2007        13    130280       1.0      13.0  130280.0   
...           ...      ...       ...       ...       ...       ...       ...   
6812821  06/04/23  14/2023        11    110020       1.0      11.0  110020.0   
6812822  21/12/23  51/2023        11    110175       1.0      11.0  110175.0   
6812823  28/02/23  09/2023        11    110020       1.0      11.0  110020.0   
6812825  10/10/23  41/2023        13    130380       1.0      13.0  130380.0   
6813431  09/01/23  02/2023        13    130406       1.0      13.0  130406.0   

         LOC_RESI  SINTOMAS  DT_SINTO  COD_OCUP  UF_INFEC  MUN_INFE  LOC_INFE  \
325201       68.0         1  17/06/03       1.0      11.0  110013.0      68.0   
1793840     372.0         1  30/01/06       2.0      11.0  110020.0     372.0   
2373846      43.0         1  24/05/06       3.0      11.0  110160.0      43.0   
2488087     372.0         1  22/12/07       1.0      11.0  110020.0     372.0   
2530206      93.0         1  30/12/06       1.0      13.0  130280.0      44.0   
...           ...       ...       ...       ...       ...       ...       ...   
6812821     932.0         1  04/04/23       3.0      11.0  110020.0     932.0   
6812822       1.0         1  18/12/23       1.0      11.0  110020.0     932.0   
6812823     934.0         1  25/02/23       6.0      11.0  110020.0     934.0   
6812825    1069.0         1  10/10/23      11.0      13.0  130380.0    1069.0   
6813431      93.0         1  01/01/23       1.0      13.0  130406.0      93.0   

         DT_EXAME  RES_EXAM  VIVAX  
325201   23/06/03         4    2.0  
1793840  01/02/06         4    2.0  
2373846  25/05/06         4    2.0  
2488087  24/12/07         4    2.0  
2530206  02/01/07         4    2.0  
...           ...       ...    ...  
6812821  06/04/23         4    2.0  
6812822  21/12/23         4    2.0  
6812823  28/02/23         4    2.0  
6812825  10/10/23         4    2.0  
6813431  09/01/23         4    2.0  

[1784858 rows x 17 columns]

In [35]:
vivax_dropped_columns_dropna = vivax_dropped_columns_dropna.replace({"UF_NOTIF": uf_dict, "UF_RESID": uf_dict})
vivax_dropped_columns_dropna

DT_NOTIF SEM_NOTI UF_NOTIF  MUN_NOTI  PAIS_RES UF_RESID  MUN_RESI  \
325201   23/06/03  26/2003       RO    110013       1.0       RO  110013.0   
1793840  01/02/06  05/2006       RO    110020       1.0       RO  110020.0   
2373846  25/05/06  21/2006       RO    110160       1.0       RO  110160.0   
2488087  24/12/07  52/2007       RO    110020       1.0       RO  110020.0   
2530206  02/01/07  01/2007       AM    130280       1.0       AM  130280.0   
...           ...      ...      ...       ...       ...      ...       ...   
6812821  06/04/23  14/2023       RO    110020       1.0       RO  110020.0   
6812822  21/12/23  51/2023       RO    110175       1.0       RO  110175.0   
6812823  28/02/23  09/2023       RO    110020       1.0       RO  110020.0   
6812825  10/10/23  41/2023       AM    130380       1.0       AM  130380.0   
6813431  09/01/23  02/2023       AM    130406       1.0       AM  130406.0   

         LOC_RESI  SINTOMAS  DT_SINTO  COD_OCUP  UF_INFEC  MUN_INFE  LOC_INFE  \
325201       68.0         1  17/06/03       1.0      11.0  110013.0      68.0   
1793840     372.0         1  30/01/06       2.0      11.0  110020.0     372.0   
2373846      43.0         1  24/05/06       3.0      11.0  110160.0      43.0   
2488087     372.0         1  22/12/07       1.0      11.0  110020.0     372.0   
2530206      93.0         1  30/12/06       1.0      13.0  130280.0      44.0   
...           ...       ...       ...       ...       ...       ...       ...   
6812821     932.0         1  04/04/23       3.0      11.0  110020.0     932.0   
6812822       1.0         1  18/12/23       1.0      11.0  110020.0     932.0   
6812823     934.0         1  25/02/23       6.0      11.0  110020.0     934.0   
6812825    1069.0         1  10/10/23      11.0      13.0  130380.0    1069.0   
6813431      93.0         1  01/01/23       1.0      13.0  130406.0      93.0   

         DT_EXAME  RES_EXAM  VIVAX  
325201   23/06/03         4    2.0  
1793840  01/02/06         4    2.0  
2373846  25/05/06         4    2.0  
2488087  24/12/07         4    2.0  
2530206  02/01/07         4    2.0  
...           ...       ...    ...  
6812821  06/04/23         4    2.0  
6812822  21/12/23         4    2.0  
6812823  28/02/23         4    2.0  
6812825  10/10/23         4    2.0  
6813431  09/01/23         4    2.0  

[1784858 rows x 17 columns]

In [32]:
#vivax_dropped_columns_dropna.to_csv('malaria_notification_data_csv/Vivax_cases.csv', index=False)

In [36]:
vivax_dropped_columns_dropna = vivax_dropped_columns_dropna.drop(columns=['PAIS_RES', 'SINTOMAS', 'DT_SINTO', 'DT_EXAME'])
vivax_dropped_columns_dropna

DT_NOTIF SEM_NOTI UF_NOTIF  MUN_NOTI UF_RESID  MUN_RESI  LOC_RESI  \
325201   23/06/03  26/2003       RO    110013       RO  110013.0      68.0   
1793840  01/02/06  05/2006       RO    110020       RO  110020.0     372.0   
2373846  25/05/06  21/2006       RO    110160       RO  110160.0      43.0   
2488087  24/12/07  52/2007       RO    110020       RO  110020.0     372.0   
2530206  02/01/07  01/2007       AM    130280       AM  130280.0      93.0   
...           ...      ...      ...       ...      ...       ...       ...   
6812821  06/04/23  14/2023       RO    110020       RO  110020.0     932.0   
6812822  21/12/23  51/2023       RO    110175       RO  110175.0       1.0   
6812823  28/02/23  09/2023       RO    110020       RO  110020.0     934.0   
6812825  10/10/23  41/2023       AM    130380       AM  130380.0    1069.0   
6813431  09/01/23  02/2023       AM    130406       AM  130406.0      93.0   

         COD_OCUP  UF_INFEC  MUN_INFE  LOC_INFE  RES_EXAM  VIVAX  
325201        1.0      11.0  110013.0      68.0         4    2.0  
1793840       2.0      11.0  110020.0     372.0         4    2.0  
2373846       3.0      11.0  110160.0      43.0         4    2.0  
2488087       1.0      11.0  110020.0     372.0         4    2.0  
2530206       1.0      13.0  130280.0      44.0         4    2.0  
...           ...       ...       ...       ...       ...    ...  
6812821       3.0      11.0  110020.0     932.0         4    2.0  
6812822       1.0      11.0  110020.0     932.0         4    2.0  
6812823       6.0      11.0  110020.0     934.0         4    2.0  
6812825      11.0      13.0  130380.0    1069.0         4    2.0  
6813431       1.0      13.0  130406.0      93.0         4    2.0  

[1784858 rows x 13 columns]

In [37]:
vivax_dropped_columns_dropna.to_csv('malaria_notification_data_csv/Vivax_cases.csv', index=False)

In [39]:
pop_df = pd.read_csv('pop_data/pop_2003-2023.csv')
pop_df = pop_df.drop(columns=['Unnamed: 0'])
pop_df

MUNIC_RES   ANO  POPULACAO
0          110001  2003      27563
1          110002  2003      79680
2          110003  2003       7478
3          110004  2003      74758
4          110005  2003      17794
...           ...   ...        ...
105771    5222005  2021      14088
105772    5222054  2021       9002
105773    5222203  2021       6451
105774    5222302  2021       5941
105775    5300108  2021    3094325

[105776 rows x 3 columns]